extensions/wikia/WallNotifications/WallNotifications.class.php

```diff
@@ -390,8 +389,7 @@ protected function sendEmails( array $watchers, WallNotificationEntity $notifica
 			if ( !empty( $mode ) && $watcher->getId() != 0 && (
 				( $mode == WALL_EMAIL_EVERY ) ||
 				( $mode == WALL_EMAIL_SINCEVISITED && empty( $this->uniqueUsers[$entityKey][$watcher->getId()] ) )
-			)) {
-
+			) && $watcher->getBoolOption( 'unsubscribed' ) !== true ) {
```

Comment:
This conditional is getting a little unwieldy.  Maybe it would be good to split this up into several, functionally grouped conditionals that issue a ```continue``` of the loop.  That would also reduce the indentation of this block, improving readability.